src/core/desktopentry.hpp

```diff
@@ -73,30 +123,111 @@ class DesktopEntry: public QObject {
+	[[nodiscard]] QBindable<QString> bindableIcon() const { return &this->bIcon; }
+	[[nodiscard]] QBindable<QString> bindableExecString() const { return &this->bExecString; }
+	[[nodiscard]] QBindable<QVector<QString>> bindableCommand() const { return &this->bCommand; }
+	[[nodiscard]] QBindable<QString> bindableWorkingDirectory() const {
+		return &this->bWorkingDirectory;
+	}
+	[[nodiscard]] QBindable<bool> bindableRunInTerminal() const { return &this->bRunInTerminal; }
+	[[nodiscard]] QBindable<QVector<QString>> bindableCategories() const {
+		return &this->bCategories;
+	}
+	[[nodiscard]] QBindable<QVector<QString>> bindableKeywords() const { return &this->bKeywords; }
+
 	// currently ignores all field codes.
 	static QVector<QString> parseExecString(const QString& execString);
 	static void doExec(const QList<QString>& execString, const QString& workingDirectory);
 
+signals:
+	void nameChanged();
+	void genericNameChanged();
+	void startupClassChanged();
+	void noDisplayChanged();
+	void commentChanged();
+	void iconChanged();
+	void execStringChanged();
+	void commandChanged();
+	void workingDirectoryChanged();
+	void runInTerminalChanged();
+	void categoriesChanged();
+	void keywordsChanged();
+
 public:
 	QString mId;
-	QString mName;
-	QString mGenericName;
-	QString mStartupClass;
-	bool mNoDisplay = false;
-	QString mComment;
-	QString mIcon;
-	QString mExecString;
-	QVector<QString> mCommand;
-	QString mWorkingDirectory;
-	bool mTerminal = false;
-	QVector<QString> mCategories;
-	QVector<QString> mKeywords;
+
+	Q_OBJECT_BINDABLE_PROPERTY(DesktopEntry, QString, bName, &DesktopEntry::nameChanged);
+	Q_OBJECT_BINDABLE_PROPERTY(
+	    DesktopEntry,
+	    QString,
+	    bGenericName,
+	    &DesktopEntry::genericNameChanged
+	);
+	Q_OBJECT_BINDABLE_PROPERTY(
+	    DesktopEntry,
+	    QString,
+	    bStartupClass,
+	    &DesktopEntry::startupClassChanged
+	);
+	Q_OBJECT_BINDABLE_PROPERTY(DesktopEntry, bool, bNoDisplay, &DesktopEntry::noDisplayChanged);
+	Q_OBJECT_BINDABLE_PROPERTY(DesktopEntry, QString, bComment, &DesktopEntry::commentChanged);
+	Q_OBJECT_BINDABLE_PROPERTY(DesktopEntry, QString, bIcon, &DesktopEntry::iconChanged);
+	Q_OBJECT_BINDABLE_PROPERTY(DesktopEntry, QString, bExecString, &DesktopEntry::execStringChanged);
+	Q_OBJECT_BINDABLE_PROPERTY(
+	    DesktopEntry,
+	    QVector<QString>,
+	    bCommand,
+	    &DesktopEntry::commandChanged
+	);
+	Q_OBJECT_BINDABLE_PROPERTY(
+	    DesktopEntry,
+	    QString,
+	    bWorkingDirectory,
+	    &DesktopEntry::workingDirectoryChanged
+	);
+	Q_OBJECT_BINDABLE_PROPERTY(
+	    DesktopEntry,
+	    bool,
+	    bRunInTerminal,
+	    &DesktopEntry::runInTerminalChanged
+	);
+	Q_OBJECT_BINDABLE_PROPERTY(
+	    DesktopEntry,
+	    QVector<QString>,
+	    bCategories,
+	    &DesktopEntry::categoriesChanged
+	);
+	Q_OBJECT_BINDABLE_PROPERTY(
+	    DesktopEntry,
+	    QVector<QString>,
+	    bKeywords,
+	    &DesktopEntry::keywordsChanged
+	);
```

Comment:
Please also keep these on single lines. Its easier to make an editor window wider than to read them spread out like this.